galaxy/api/v2/pagination.py

```diff
@@ -0,0 +1,54 @@
+# (c) 2012-2019, Ansible by Red Hat
+#
+# This file is part of Ansible Galaxy
+#
+# Ansible Galaxy is free software: you can redistribute it and/or modify
+# it under the terms of the Apache License as published by
+# the Apache Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# Ansible Galaxy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# Apache License for more details.
+#
+# You should have received a copy of the Apache License
+# along with Galaxy.  If not, see <http://www.apache.org/licenses/>.
+
+from rest_framework import pagination
+from rest_framework.utils.urls import remove_query_param, replace_query_param
+
+
+class CustomPagination(pagination.PageNumberPagination):
```

Comment:
Why did you implement custom pagination class?

Reply:
api/v1 gives multiple next and previous links with a hardcoded removal of ‘api/v1’:
```
"next": "/repositories/?page=2",
"next_link": "/api/v1/repositories/?page=2",
"previous": null,
"previous_link": null,
```
api/v2 should be more standard unless otherwise needed:
```
"next": "/api/v2/collections/1/versions/?page=2",
"previous": null,
```

This v2 pagination class can be moved to the global default, and v1 endpoints changed to use the old pagination class - but should be done in separate PR

Reply:
Custom pagination class logic was narrowed to just custom page size